exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -68,5 +79,151 @@ func (prwe *prwExporter) shutdown(context.Context) error {
+func (prwe *prwExporter) handleScalarMetric(tsMap map[string]*prompb.TimeSeries, metric *otlp.Metric) error {
+
+	mType := metric.MetricDescriptor.Type
+
+	switch mType {
+	// int points
+	case otlp.MetricDescriptor_MONOTONIC_INT64, otlp.MetricDescriptor_INT64:
+		if metric.Int64DataPoints == nil {
+			return errors.New("nil data point field in metric" + metric.GetMetricDescriptor().Name)
+		}
+
+		for _, pt := range metric.Int64DataPoints {
+
+			// create parameters for addSample
+			name := getPromMetricName(metric.GetMetricDescriptor(), prwe.namespace)
+			labels := createLabelSet(pt.GetLabels(), nameStr, name)
+			sample := &prompb.Sample{
+				Value: float64(pt.Value),
+				// convert ns to ms
+				Timestamp: convertTimeStamp(pt.TimeUnixNano),
+			}
+
+			addSample(tsMap, sample, labels, metric.GetMetricDescriptor().GetType())
+		}
+		return nil
+
+	// double points
+	case otlp.MetricDescriptor_MONOTONIC_DOUBLE, otlp.MetricDescriptor_DOUBLE:
+		if metric.DoubleDataPoints == nil {
+			return errors.New("nil data point field in metric" + metric.GetMetricDescriptor().Name)
+		}
+		for _, pt := range metric.DoubleDataPoints {
+
+			// create parameters for addSample
+			name := getPromMetricName(metric.GetMetricDescriptor(), prwe.namespace)
+			labels := createLabelSet(pt.GetLabels(), nameStr, name)
+			sample := &prompb.Sample{
+				Value:     pt.Value,
+				Timestamp: convertTimeStamp(pt.TimeUnixNano),
+			}
+
+			addSample(tsMap, sample, labels, metric.GetMetricDescriptor().GetType())
+		}
+		return nil
+	}
+
+	return errors.New("invalid metric type: wants int or double data points")
+}
+
+// export sends a Snappy-compressed WriteRequest containing TimeSeries to a remote write endpoint in order
+func (prwe *prwExporter) export(ctx context.Context, tsMap map[string]*prompb.TimeSeries) error {
+	//Calls the helper function to convert the TsMap to the desired format
+	req, err := wrapTimeSeries(tsMap)
+	if err != nil {
+		return err
+	}
+
+	//Uses proto.Marshal to convert the WriteRequest into bytes array
+	data, err := proto.Marshal(req)
+	if err != nil {
+		return err
+	}
+	buf := make([]byte, len(data), cap(data))
+	compressedData := snappy.Encode(buf, data)
+
+	//Create the HTTP POST request to send to the endpoint
+	httpReq, err := http.NewRequest("POST", prwe.endpointURL.String(), bytes.NewReader(compressedData))
+	if err != nil {
+		return err
+	}
+
+	// Add necessary headers specified by:
+	// https://cortexmetrics.io/docs/apis/#remote-api
+	httpReq.Header.Add("Content-Encoding", "snappy")
+	httpReq.Header.Set("Content-Type", "application/x-protobuf")
+	httpReq.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+
+	httpReq = httpReq.WithContext(ctx)
+
+	_, cancel := context.WithTimeout(context.Background(), prwe.client.Timeout)
+	defer cancel()
+
+	httpResp, err := prwe.client.Do(httpReq)
+	if err != nil {
+		return err
+	}
+
+	if httpResp.StatusCode/100 != 2 {
```

Comment:
nit: I noticed the OTel-Go exporter tests for 200 (`http.StatusOK`) exactly. Do you think the other values really matter? The other value I'd maybe expect is 202. Do you think the other 200 codes matter?

Reply:
I am not sure about other code. The reference here is the Prometheus remote write implementation:https://github.com/prometheus/prometheus/blob/2c4a4548a8382f7c8966dbfda37f34c43151e316/storage/remote/client.go#L187